specutils/spectra/spectrum1d.py

```diff
@@ -252,17 +252,43 @@ def __getitem__(self, item):
                 if len(item) == len(self.flux.shape) or item[0] == Ellipsis:
                     spec_item = item[-1]
                     if not isinstance(spec_item, slice):
+                        if isinstance(item, u.Quantity):
+                            raise ValueError("Indexing on single spectral axis "
+                                             "values is not currently allowed, "
+                                             "please use a slice.")
                         spec_item = slice(spec_item, spec_item+1, None)
                         item = item[:-1] + (spec_item,)
                 else:
                     # Slicing on less than the full number of axes means we want
                     # to keep the whole spectral axis
                     spec_item = slice(None, None, None)
+            elif isinstance(item, slice) and (isinstance(item.start, u.Quantity) or
+                    isinstance(item.stop, u.Quantity)):
+                # We only allow slicing with world coordinates along the spectral
+                # axis for now
+                for attr in ("start", "stop"):
+                    if getattr(item, attr) is None:
+                        continue
+                    if not getattr(item, attr).unit.is_equivalent(u.AA,
+                            equivalencies=u.spectral()):
+                        raise ValueError("Slicing with world coordinates is only"
+                                         " enabled for spectral coordinates.")
+                        break
+                spec_item = item
             else:
                 # Slicing with a single integer or slice uses the leading axis,
                 # so we keep the whole spectral axis, which is last
                 spec_item = slice(None, None, None)
 
+            if (isinstance(spec_item.start, u.Quantity) or
+                    isinstance(spec_item.stop, u.Quantity)):
+                temp_spec = self._spectral_slice(spec_item)
```

Comment:
I updated the upper bound to shift to the next lower spectral axis value if the input upper bound is an exact match to a spectral axis value.

While I agree that it would be nice to take be able to do `spectrum[SpectralRegion(...)]`, it raises the same question about how we want to handle the upper bound. The case of just putting spectral values in brackets like a normal array slice (the case I just updated) seems clear and matches user expectations, but do we really want to be handling an actual `SpectralRegion` object differently if it's input to an array slice vs input to `extract_region`? I'm not sure what the user expectation would be there.

Reply:
I kind of want to leave the `SpectralRegion` case to be handled explicitly by `extract_region` rather than hiding `extract_region` behind the __getitem__ behavior, but that might be laziness talking.

Reply:
Hmm.  I see your point here. It *does* fit with "There should be one-- and preferably only one --obvious way to do it." (see `import this` if you don't know what I'm talking about).

And this can always be implemented later as a follow-on PR, so on "don't let the perfect be the enemy of the good", lets drop my `spectrum[SpectralRegion(...)]` suggestion then - I can make an issue if I really care about it.